compute/cloud-client/src/main/java/compute/reservation/CreateTemplateNotConsumeReservation.java

```diff
@@ -0,0 +1,111 @@
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class CreateTemplateNotConsumeReservation {
+  public static void main(String[] args)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // TODO(developer): Replace these variables before running the sample.
+    // Project ID or project number of the Cloud project you want to use.
+    String projectId = "YOUR_PROJECT_ID";
+    // Name of the template you want to query.
+    String templateName = "YOUR_INSTANCE_TEMPLATE_NAME";
+
+    createTemplateNotConsumeReservation(projectId, templateName);
+  }
+
+
+  // Create a template that explicitly doesn't consume any reservations.
+  public static void createTemplateNotConsumeReservation(String projectId, String templateName)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    try (InstanceTemplatesClient instanceTemplatesClient = InstanceTemplatesClient.create()) {
+
+      String machineType = "e2-standard-4";
+      String sourceImage = "projects/debian-cloud/global/images/family/debian-11";
+
+      // The template describes the size and source image of the boot disk
+      // to attach to the instance.
+      AttachedDisk attachedDisk = AttachedDisk.newBuilder()
+          .setInitializeParams(AttachedDiskInitializeParams.newBuilder()
+              .setSourceImage(sourceImage)
+              .setDiskType("pd-balanced")
+              .setDiskSizeGb(250).build())
+          .setAutoDelete(true)
+          .setBoot(true).build();
+
+      // The template connects the instance to the `default` network,
+      // without specifying a subnetwork.
+      NetworkInterface networkInterface = NetworkInterface.newBuilder()
+          .setName("global/networks/default")
+          // The template lets the instance use an external IP address.
+          .addAccessConfigs(AccessConfig.newBuilder()
+              .setName("External NAT")
+              .setType(AccessConfig.Type.ONE_TO_ONE_NAT.toString())
+              .setNetworkTier(AccessConfig.NetworkTier.PREMIUM.toString()).build()).build();
+
+      // Set reservation affinity to "none"
+      ReservationAffinity reservationAffinity =
+          ReservationAffinity.newBuilder()
+              .setConsumeReservationType(NO_RESERVATION.toString())
+              .build();
+
+      InstanceProperties instanceProperties = InstanceProperties.newBuilder()
+          .addDisks(attachedDisk)
+          .setMachineType(machineType)
+          .setReservationAffinity(reservationAffinity)
+          .addNetworkInterfaces(networkInterface).build();
+
+      InsertInstanceTemplateRequest insertInstanceTemplateRequest = InsertInstanceTemplateRequest
+          .newBuilder()
+          .setProject(projectId)
+          .setInstanceTemplateResource(InstanceTemplate.newBuilder()
+              .setName(templateName)
+              .setProperties(instanceProperties).build()).build();
```

Comment:
nit: consider breaking two consecutive build() calls to separate lines to increase readability.

```suggestion
              .setProperties(instanceProperties).build())
          .build();
```